snuba/query/processors/prewhere.py

```diff
@@ -55,9 +64,60 @@ def process_query(self, query: Query, request_settings: RequestSettings,) -> Non
         )
         if prewhere_candidates:
             prewhere_conditions = [cond for _, cond in prewhere_candidates][
-                : self.__max_prewhere_conditions
+                :max_prewhere_conditions
             ]
             query.set_conditions(
                 list(filter(lambda cond: cond not in prewhere_conditions, conditions))
             )
         query.set_prewhere(prewhere_conditions)
+
+
+class CustomPrewhereProcessor(PrewhereProcessor):
+    def __init__(
+        self,
+        custom_candidates: Sequence[str],
+        max_prewhere_conditions: Optional[int] = None,
+    ) -> None:
+        super().__init__(max_prewhere_conditions)
+        self.__custom_candidates = custom_candidates
+
+    def _get_prewhere_keys(self, query: Query) -> Sequence[str]:
+        candidates = query.get_data_source().get_prewhere_candidates()
+
+        custom_projects_config = state.get_config("prewhere_custom_key_projects", "[]")
+        custom_projects = json.loads(custom_projects_config)
+
+        condition = query.get_condition_from_ast()
+
+        def is_project_condition(node: Expression) -> bool:
+            return (
+                is_in_condition(node)
+                and isinstance(node, FunctionCall)
```

Comment:
Isn't this already enforced by `is_in_condition`? Is this required to satisfy the type checker?

Reply:
the isinstance is just for the type cheker